xdsl/irdl/attributes.py

```diff
@@ -405,7 +399,7 @@ def constr(self) -> AttrConstraint[AttributeCovT]:
     AttrConstraint[AttributeInvT]
     | AttributeInvT
     | type[AttributeInvT]
-    | "TypeForm[AttributeInvT]"
+    | TypeForm[AttributeInvT]
```

Comment:
This seems unfortunate, but python 3.14 does not like trying to union a string and a type. I'm not entirely sure what the ramifications of making this change are.

Reply:
There was a regression in 3.14: https://github.com/xdslproject/xdsl/issues/5396